onnxruntime/test/fuzzing/src/test.cpp

```diff
@@ -265,143 +239,170 @@ int processCommandLine(int argc, char* argv[], runtimeOpt& opt)
+        so.AddConfigEntry(kOrtSessionOptionsConfigSaveModelFormat, "ORT");
+        Ort::Session session(env, model_file.c_str(), so);
+      } else if (model_file.substr(model_file.find_last_of(L".") + 1) != L"ort") {
+        Logger::testLog << L"Input file name extension is not 'onnx' or 'ort' " << Logger::endl;
+        return 1;
+      }
+      size_t num_bytes = std::filesystem::file_size(ort_model_file);
+      std::vector<char> model_data(num_bytes);
+      std::ifstream ortModelStream(ort_model_file, std::ifstream::in | std::ifstream::binary);
+      ortModelStream.read(model_data.data(), num_bytes);
+      ortModelStream.close();
+      // Currently mutations are generated by using XOR of a byte with the preceeding byte at a time.
+      // Other possible ways may be considered in future, for example swaping two bytes randomly at a time.
+      Logger::testLog << "Starting Test" << Logger::endl;
+      for (size_t& i = run_stats.iteration; i < num_bytes - 1; i++) {
+        char tmp = model_data[i];
+        model_data[i] ^= model_data[i + 1];
+        try {
+          Logger::testLog << "Starting Test iteration: " << i << Logger::endl;
+          OnnxPrediction predict(model_data, env);
+          predict.SetupInput(GenerateDataForInputTypeTensor, 0);
+          predict.RunInference();
+          run_stats.num_successful_runs++;
+          Logger::testLog << "Completed Test iteration: " << i << Logger::endl;
+        } catch (...) {
+          fuzz_handle_exception(run_stats);
         }
+        model_data[i] = tmp;
       }
     }
-    else
-    {
-      throw std::exception("Unable to initialize the Onnx model in memory");
-    }
+    Logger::testLog << "Ending Test" << Logger::endl;
 
+    if (user_opt.stress) {
+      Logger::testLog.enable();
+    }
+    size_t toal_num_exception = run_stats.num_unknown_exception + run_stats.num_std_exception + run_stats.num_ort_exception;
+    Logger::testLog << L"Total number of exceptions: " << toal_num_exception << Logger::endl;
+    Logger::testLog << L"Number of Unknown exceptions: " << run_stats.num_unknown_exception << Logger::endl;
+    Logger::testLog << L"Number of ort exceptions: " << run_stats.num_ort_exception << Logger::endl;
+    Logger::testLog << L"Number of std exceptions: " << run_stats.num_std_exception << Logger::endl;
+    Logger::testLog << L"Number of unique errors: " << werr_stream_buf.get_unique_errors() << L"\n";
+
+    if (user_opt.stress) {
+      Logger::testLog.disable();
+      Logger::testLog.flush();
+    }
     return 0;
-  }
-  catch(const Ort::Exception& ortException)
-  {
-    Logger::testLog<< L"onnx runtime exception: " << ortException.what() << Logger::endl;
-  }
-  catch(const std::exception& e)
-  {
-    Logger::testLog<< L"standard exception: " << e.what() << Logger::endl;
-  }
-  catch(...)
-  {
-    Logger::testLog<< L"Something Went very wrong: " << Logger::endl;
+  } catch (const Ort::Exception& ort_exception) {
```

Comment:
Should this return a non-zero code if there are non-ORT exceptions? Wondering whether the CI would fail due to a fuzz testing error if we don't return that info.

Reply:
There is a "return 1" at line #408. Doesn't it return non-zero? what am I missing?

Reply:
Sorry - github/codeflow mismatch. was referring to the line 399 that has 'return 0' so any errors from the fuzz testing loop with the try/catch don't result in a non-zero return.